apps/src/lib/ui/SchoolInfoConfirmationDialog.jsx

```diff
@@ -15,80 +15,96 @@ export const styles = {
-    isOpen: PropTypes.bool.isRequired
+    scriptData: PropTypes.shape({
+      formUrl: PropTypes.string.isRequired,
+      authTokenName: PropTypes.string.isRequired,
+      authTokenValue: PropTypes.string.isRequired,
+      existingSchoolInfo: PropTypes.shape({
+        id: PropTypes.number,
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        full_address: PropTypes.string
+      }).isRequired
+    }).isRequired,
+    onClose: PropTypes.func,
+    isOpen: PropTypes.bool
   };
 
   constructor(props) {
     super(props);
     this.state = {
       showSchoolInterstitial: false,
-      schoolName: props.schoolName,
-      isOpen: true
+      schoolName: props.scriptData.existingSchoolInfo.school_name,
+      isOpen: props.isOpen || true
     };
   }
 
-  static defaultProps = {
-    onUpdate: () => {},
-    schoolName: ''
-  };
+  handleClickYes = () => {
+    const {authTokenName, authTokenValue} = this.props.scriptData;
 
-  componentDidMount() {
-    const {schoolName} = this.state;
-    if (!schoolName && schoolName.length > 0) {
-      fetch('/dashboardapi/v1/users/me/school_name')
-        .then(response => response.json())
-        .then(data => {
-          this.setState({
-            schoolName: data.school_name
-          });
-        })
-        .catch(error => this.setState({error}));
-    }
-  }
+    console.log(authTokenName, authTokenValue);
+
+    fetch(
+      `/api/v1/user_school_infos/${
+        this.props.scriptData.existingSchoolInfo.id
+      }/update_last_confirmation_date`,
+      {method: 'PATCH', headers: {[authTokenName]: authTokenValue}}
+    )
+      .then(() => this.props.onClose())
+      .catch(error => {
+        console.log('The Error =>', error);
+        this.setState({error});
+      });
+  };
 
-  handleUpdateClick = () => {
-    this.setState({showSchoolInterstitial: true});
+  handleClickUpdate = () => {
+    console.log('===>I have been called');
+    const {authTokenName, authTokenValue} = this.props.scriptData;
+    fetch(
+      `/api/v1/user_school_infos/${
+        this.props.scriptData.existingSchoolInfo.id
+      }/update_end_date`,
+      {method: 'PATCH', headers: {[authTokenName]: authTokenValue}}
+    )
+      .then(() => this.setState({showSchoolInterstitial: true}))
+      .catch(() => {});
   };
 
-  renderInitialContent() {
-    const {onConfirm} = this.props;
+  renderInitialContent = () => {
     const {schoolName} = this.state;
     return (
       <Body>
         <div>
-          <p>
-            {i18n.schoolInfoDialogDescription()} {schoolName}
-          </p>
+          <p>{i18n.schoolInfoDialogDescription({schoolName})}</p>
         </div>
         <Button
           text={i18n.schoolInfoDialogUpdate()}
           color={Button.ButtonColor.blue}
-          onClick={this.handleUpdateClick}
+          onClick={this.handleClickUpdate}
+          id="first-button"
```

Comment:
nit: make these ids a bit more descriptive (i.e. update-button or similar)

Reply:
Thanks for the catch, I forgot to update.[](url)